src/main/java/com/st/lms/service/AdminService.java

```diff
@@ -0,0 +1,338 @@
+package com.st.lms.service;
+
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.st.lms.dao.AuthorDao;
+import com.st.lms.dao.BookDao;
+import com.st.lms.dao.BookLoansDao;
+import com.st.lms.dao.BorrowerDao;
+import com.st.lms.dao.LibBranchDao;
+import com.st.lms.dao.PublisherDao;
+import com.st.lms.dto.BkAuthPubDTO;
+import com.st.lms.models.Author;
+import com.st.lms.models.Book;
+import com.st.lms.models.BookLoans;
+import com.st.lms.models.Borrower;
+import com.st.lms.models.LibraryBranch;
+import com.st.lms.models.Publisher;
+
+@Service
+public class AdminService {
+	
+	@Autowired
+	private BookDao bookDao;
+	@Autowired
+	private PublisherDao publisherDao;
+	@Autowired
+	private BookLoansDao bookLoansDao;
+	@Autowired
+	private AuthorDao authorDao;
+	@Autowired
+	private BorrowerDao borrowerDao;
+	@Autowired
+	private LibBranchDao libraryBranchDao;
+
+	public boolean bookTitleExists(String bookTitle) {
+		List<Book> books = bookDao.findAll();
+		
+		for(Book b : books) {
+			if(b.getTitle().equals(bookTitle))
+				return true;
+		}
+		return false;
+	}
+	
+	public void addBook(String title, int authId, int pubId) {
+		Book b = new Book();
+		b.setTitle(title);
+		b.setAuthorId(authId);
+		b.setPubId(pubId);
+
+		bookDao.save(b);
+		System.out.println("Add Book Success!");
+	}
+	
+	public Book getBook(int bookId) {
+		//Book book = null;
+		//book = bookDao.findById(bookId).get();
+		//return book;
+		
+		//prevents having to throw NoSuchElementException
+		Optional<Book> book;
+		book = bookDao.findById(bookId);
+		return book.isPresent() ? book.get() : null;
+	}
+	
+	public List<Book> getAllBooks() {
+		List<Book> books = null;
+		books = bookDao.findAll();
+		return books;
+	}
+	
+	public BkAuthPubDTO getBookWithAuthAndPub(int bookId) {
+		BkAuthPubDTO obj = null;
+		List<Book> books = bookDao.findAll();
+		Author author;
+		Publisher publisher;
+		
+		for(Book b : books) {

```

Comment:
Just for future reference, you could write this function using in functional programming. There would probably be not difference in speed, but it would make it easier to read.